nomad/periodic.go

```diff
@@ -0,0 +1,529 @@
+package nomad
+
+import (
+	"container/heap"
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+const (
+	// The string appended to the periodic jobs ID when launching derived
+	// instances of it.
+	JobLaunchSuffix = "/periodic-"
+)
+
+// PeriodicRunner is the interface for tracking and launching periodic jobs at
+// their periodic spec.
+type PeriodicRunner interface {
```

Comment:
I'd change this around such that there is no `PeriodicRunner` interface, but instead of passing the server into `PeriodicDispatch` you pass that in as an interface. e.g. the `EvalDispatch` interface. That way you can test the PeriodicDispatch more thoroughly by mocking the server itself. There is no other expected implementation of PeriodicRunner